homeassistant/components/komfovent/climate.py

```diff
@@ -0,0 +1,84 @@
+"""Ventilation Units from Komfovent integration."""
+from __future__ import annotations
+
+import komfovent_api
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, MANUFACTURER, PARAM_HOST, PARAM_PASSWORD, PARAM_USERNAME
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Komfovent unit control."""
+    conf_host = str(entry.data[PARAM_HOST])
+    conf_username = str(entry.data[PARAM_USERNAME])
+    conf_password = str(entry.data[PARAM_PASSWORD])
+    _, credentials = komfovent_api.get_credentials(
+        conf_host, conf_username, conf_password
+    )
+
+    result, settings = await komfovent_api.get_settings(credentials)
+    if result == komfovent_api.KomfoventConnectionResult.SUCCESS:
```

Comment:
I assume by in `integration setup` you mean the config flow (please correct me if i am wrong)? I have that function call there as well, here i am just fetching the information for the entity setup. The check is just for a weird fluke if something like internet connection drops out. Or do you want me to raise an exception here instead of just not calling add_entities? Sorry for the rookie question, but i am still not exactly familiar with HASS workings.